mis_builder/models/mis_builder.py

```diff
@@ -0,0 +1,865 @@
+                            string='Name'),
+        'model_id': fields.many2one('ir.model', required=True,
+                                    string='Model'),
+        'field_ids': fields.many2many('ir.model.fields', required=True,
+                                      string='Fields to fetch'),
+        'field_names': fields.function(_get_field_names, type='char',
+                                       string='Fetched fields name',
+                                       store={'mis.report.query':
+                                              (lambda self, cr, uid, ids, c={}:
+                                               ids, ['field_ids'], 20), }),
+        'date_field': fields.many2one('ir.model.fields', required=True,
+                                      string='Date field',
+                                      domain=[('ttype', 'in',
+                                               ('date', 'datetime'))]),
+        'domain': fields.char(string='Domain'),
+        'report_id': fields.many2one('mis.report', string='Report',
+                                     ondelete='cascade'),
+    }
+
+    _order = 'name'
+
+    def _check_name(self, cr, uid, ids, context=None):
+        for record_name in self.read(cr, uid, ids, ['name']):
+            if not _is_valid_python_var(record_name['name']):
+                return False
+        return True
+
+    _constraints = [
+        (_check_name, 'The name must be a valid python identifier', ['name']),
+    ]
+
+
+class mis_report(orm.Model):
+
+    """ A MIS report template (without period information)
+
+    The MIS report holds:
+    * an implicit query fetching all the account balances;
+      for each account, the balance is stored in a variable named
+      bal_{code} where {code} is the account code
+    * an implicit query fetching all the account balances solde;
+      for each account, the balance solde is stored in a variable named
+      bals_{code} where {code} is the account code
+    * a list of explicit queries; the result of each query is
+      stored in a variable with same name as a query, containing as list
+      of data structures populated with attributes for each fields to fetch
+    * a list of KPI to be evaluated based on the variables resulting
+      from the balance and queries
+    """
+
+    _name = 'mis.report'
+
+    _columns = {
+        'name': fields.char(size=32, required=True,
+                            string='Name', translate=True),
+        'description': fields.char(required=False,
+                                   string='Description', translate=True),
+        'query_ids': fields.one2many('mis.report.query', 'report_id',
+                                     string='Queries'),
+        'kpi_ids': fields.one2many('mis.report.kpi', 'report_id',
+                                   string='KPI\'s'),
+    }
+
+    # TODO: kpi name cannot be start with query name
+
+    def create(self, cr, uid, vals, context=None):
+        # TODO: explain this
```

Comment:
see : https://github.com/odoo/odoo/blob/8.0/addons/account/account_bank_statement.py#L36
it is use to initialize sequence